src/mapping/Mapping.cpp

```diff
@@ -225,7 +225,7 @@ void Mapping::scaleConsistentMapping(const Eigen::VectorXd &input, Eigen::Vector
   Eigen::Map<Eigen::MatrixXd> outputValuesMatrix(output.data(), valueDimensions, output.size() / valueDimensions);
 
   // Scale in each direction
-  Eigen::VectorXd scalingFactor = integralInput.array() / integralOutput.array();
+  const Eigen::VectorXd scalingFactor = integralInput.binaryExpr(integralOutput, [](double lhs, double rhs) { return (rhs == 0.0) ? 0.0 : (lhs / rhs); });
```

Comment:
I'm a bit in doubt about this: When having no flux, then the solution of the applied mapping here is undefined. Why is zero the desired solution? Shouldn't we abort at this point or at least give a warning? What about 
```suggestion
  const Eigen::VectorXd scalingFactor = integralInput.binaryExpr(integralOutput, [](double lhs, double rhs) { return (rhs == 0.0) ? lhs : (lhs / rhs); });
```

Reply:
`0.0` seems to be the right value if the input is zero everywhere, which is the bug we ran into.
It could, indeed, however also be that the integral is zero, but the values are not.

```suggestion
  const Eigen::VectorXd scalingFactor = integralInput.binaryExpr(integralOutput, [](double lhs, double rhs) { return (rhs == 0.0) ? 1.0 : (lhs / rhs); });
```

??

I don't understand the `lhs` suggestion. It's "only" the scaling factor, right? The values still get multiplied below.